migration-engine/connectors/migration-connector/src/destructive_changes_checker.rs

```diff
@@ -16,6 +16,9 @@ where
 
     /// Check destructive changes resulting of reverting the provided migration.
     async fn check_unapply(&self, database_migration: &T) -> ConnectorResult<DestructiveChangeDiagnostics>;
+
+    /// Check the database migration without performing any IO.
```

Comment:
By "check" do you mean "validate this migration is correct"?

Reply:
I'll clarify that. It's a bit clearer in context of the trait documentation above: "/// Implementors of this trait are responsible for checking whether a migration could lead to data loss."